app/src/main/java/com/guruthedev/instagram/ui/fragments/auth/signUp/SignUpFragment.kt

```diff
@@ -43,13 +43,13 @@ class SignUpFragment : Fragment() {
         }
         viewModel.errorLiveData.observe(viewLifecycleOwner) { signUpError ->
             when (signUpError.signUpTypeError) {
-                SignUpErrorType.ERROR_EMPTY_FULL_NAME -> showError(getString(R.string.full_name_toast_message))
-                SignUpErrorType.ERROR_EMPTY_USERNAME -> showError(getString(R.string.user_name_toast_message))
-                SignUpErrorType.ERROR_EMPTY_EMAIL -> showError(getString(R.string.email_toast_message))
-                SignUpErrorType.ERROR_EMPTY_PASSWORD -> showError(getString(R.string.password_toast_message))
+                SignUpErrorType.ERROR_EMPTY_FULL_NAME -> requireContext().showToast(getString(R.string.error_empty_full_name))
```

Comment:
can we use a scope function here like 
`with(requireContext()) {
 when (signUpError.signUpTypeError) {
 SignUpErrorType.ERROR_EMPTY_FULL_NAME -> showToast(getString(R.string.error_empty_full_name))
...
...
}
}`